duet/views/mobile_ios_country.view.lkml

```diff
@@ -0,0 +1,368 @@
+view: mobile_ios_country {
+  derived_table: {
+    sql: WITH last_updated AS (
+        -- this table has rows that lag ~3 days plus/minus several days
+        SELECT DISTINCT
+          max(date) AS ios_store_updated,
+          -- We need a full week to see activated client; ignore the current day of partial data
+          date_sub(current_date(), INTERVAL 7 + 1 day) AS latest_date
+        FROM
+          `moz-fx-data-marketing-prod.apple_app_store.metrics_by_storefront`
+        WHERE
+          date >= date_sub(current_date(), INTERVAL 28 day)
+      ),
+      period AS (
+        SELECT
+          ios_store_updated,
+          latest_date,
+          date_sub(latest_date,
+            INTERVAL {% parameter.history_days %} * ({% parameter.period_offset %} + 1) - 1  day
+          ) AS start_date,
+          date_sub(latest_date,
+            INTERVAL {% parameter.history_days %} * {% parameter.period_offset %} day
+          ) AS end_date
+        FROM
+          last_updated
+      ),
+      apple_storefront_metrics
+      AS
+        (
+          SELECT
+            date,
+            app_name,
+            CASE
+            WHEN
+              storefront LIKE '%St.%'
+            THEN
+              REPLACE(storefront, 'St.', 'Saint')
+            WHEN
+              storefront = 'China mainland'
+            THEN
```

Comment:
What do you think about pulling this big thing out in to a UDF? That would simplify things here, and make it usable for other analyses that need to join apple and telemetry data.

Reply:
That seems fine, the logic is kind of wonky. I didn't want to go through the hassle when I was building this since it would have taken longer to get to the end state (getting the udf deployed from bigquery-etl).

Reply:
I understand, but it will give a better guarantee of correctness here. The existing query is hard to review because of the length.

Reply:
Okay, I filed https://github.com/mozilla/bigquery-etl/issues/2068.

Reply:
Resolved